Enable user to import a site that is not using WordPress as platform

gabrielcaires opened the pull request:
Related to #94914

Proposed Changes

* Add `other-platform-detected-import` step to site-migration and migration flow
* Detect if site is not using WordPress as platform and redirect to the new step
* Refactor on 'useCredentialsForm` to simplify the code and accommodate the changes better.


## Why are these changes being made?
* We have observed many users trying to migrate sites that are not using WordPress; this PR aims to capture them and redirect them to the correct step.


## Testing Instructions

**Scenario: Importing content**
* Go to `/setup/migration` and follow the steps until you reach the credential step
* Fill the "Site address" field with a site that is using one of the platforms we support importing  (You can see a list of platforms we are supporting [here](https://github.com/Automattic/wp-calypso/blob/20eeb0b35e3ad476a8bdcbabee2fe258a8e12ac7/client/blocks/import/util.ts#L18-L33)
*  Check if you were redirected to the new step. 
* Click on Import and check if you were redirected to the proper import screen based

**Scenario: Importing content**
* Go to `/setup/migration` and follow the steps until you reach the credential step
* Fill the "Site address" field with a site that is using one of the platforms we support importing  (You can see a list of platforms we are supporting [here](https://github.com/Automattic/wp-calypso/blob/20eeb0b35e3ad476a8bdcbabee2fe258a8e12ac7/client/blocks/import/util.ts#L18-L33)
* Check if you were redirected to the new step. 
* Click on "I need help" 
* Check if you were redirected to the `We'll take it from here!` step

> [!NOTE]  
> We are planning to update the `We'll take it from here!` step to be more contextual by case, it will be done in further PRs.

Please test, the same scenarios are applicable to the `/setup/site-migration` flow



<img width="815" alt="image" src="https://github.com/user-attachments/assets/8fa27d93-de0b-46da-ba87-2596f66b3942">


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

gabrielcaires committed:
Add step

gabrielcaires committed:
Add logic to support handle sites not using WordPress

gabrielcaires committed:
Rename mutation to describe better the intent

gabrielcaires committed:
ADd support to receive sites not using WordPress

gabrielcaires committed:
Show platform name

gabrielcaires committed:
Introduce step on migration flow

gabrielcaires committed:
Introduce step on site-migration flow

gabrielcaires committed:
Remove unsued handleSubmit

gabrielcaires committed:
Remove scanning component

gabrielcaires committed:
Remove scan test

gabrielcaires committed:
Add missing type

gabrielcaires commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/hooks/use-credentials-form.ts:
```diff
@@ -60,68 +64,44 @@ export const useCredentialsForm = ( onSubmit: ( siteInfo?: UrlData ) => void ) =
-		return translate( 'Continue' );
-	}, [ isSubmitting, canBypassVerification, isEnglishLocale, translate ] );
 
-	useEffect( () => {
```
The hooks is not necessary, we can capture the success directly on the submit handler.

gabrielcaires commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/hooks/use-credentials-form.ts:
```diff
@@ -142,10 +122,11 @@ export const useCredentialsForm = ( onSubmit: ( siteInfo?: UrlData ) => void ) =
-		accessMethod: watch( 'migrationType' ),
+		accessMethod,
 		isBusy,
-		getContinueButtonText,
```
I extracted it from the hook because it is a UI responsibility instead of a hook responsibility

gabrielcaires commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-other-platform-detected-import/index.tsx:
```diff
@@ -0,0 +1,107 @@
+		isFetching: isAnalyzingUrl,
+		data: siteInfo,
+		isSuccess,
+	} = useAnalyzeUrlQuery( from, shouldAnalyzeUrl );
```
It only runs the site analyzes when the platform is not available on the URL.

It is necessary because when the user uses the back button on some importing page, we lose the platform name. 
So, instead of updating all importer options (WPCOM and WPorg), I just made the step analysis again.

andres-blanco commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/hooks/use-credentials-form.ts:
```diff
@@ -1,41 +1,45 @@
 import { CredentialsFormData } from '../types';
 import { useFormErrorMapping } from './use-form-error-mapping';
-import { useSiteMigrationCredentialsMutation } from './use-site-migration-credentials-mutation';
+import { useRequestAutomatedMigration as useRequestAutomatedMigration } from './use-request-automated-migration';
```
```suggestion
import { useRequestAutomatedMigration } from './use-request-automated-migration';
```

gabrielcaires replied:
AI BUG fixed here c4bb9d595c455d18c28bc3de019891f623b0f69b

Imran92 commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-other-platform-detected-import/index.tsx:
```diff
@@ -0,0 +1,107 @@
+							id="site-migration-credentials-header"
+							headerText={ translate( "Looks like there's been a mix-up" ) }
+							subHeaderText={ translate(
+								'Our migration service is for WordPress sites. But don’t worry — our {{strong}}%(platform)s{{/strong}} import tool is ready to bring your content to WordPress.com.',
```
Should we add a placeholder comment?

gabrielcaires replied:
Fixed here https://github.com/Automattic/wp-calypso/pull/95246/commits/ccf0903a86fa25293b3835c68c767b708421550c#diff-79efeb095b567cb4da602b05c948e66ab26c13be9f32cebb278928353db7b56fR64

Imran92 commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-other-platform-detected-import/index.tsx:
```diff
@@ -0,0 +1,107 @@
+					) : (
+						<FormattedHeader
+							id="site-migration-credentials-header"
+							headerText={ translate( "Looks like there's been a mix-up" ) }
```
As the same text is being used in DocumentHead and here, should we declare it just once in a variable above and use that variable in both of these places?

gabrielcaires replied:
Fixed here 23fdaae6d35a5a1b756cd1bfc10219d36776edc6

Imran92 commented on client/landing/stepper/declarative-flow/migration/helpers/index.ts:
```diff
@@ -32,12 +43,21 @@ export const goToImporter = ( {
+	ref,
 }: GoToImporterParams ) => {
-	const backToFlow = backToStep ? getFlowPath( backToStep?.slug ) : undefined;
+	const backToFlowURL = backToFlow || ( backToStep ? getFlowPath( backToStep?.slug ) : undefined );
```
A bit too many checks in one line, should we break it down in a couple of lines for maybe a bit better readability?

gabrielcaires replied:
Fixed here bc696c9ea1c05bde6d6b6ce1e69ee4f772f59950

Imran92 reviewed (commented):
Hi @gabrielcaires ! Nice work on this. Most of the things works nicely.

However, I've left some optional comments on code and have noticed some hiccups in the following scenarios -

1. The back button doesn't work on the this page -

<img width="1404" alt="Screenshot 2024-10-15 at 7 04 26 AM" src="https://github.com/user-attachments/assets/d6428bb7-fd8d-487c-92d9-2e533cef1cb5">

2. We are showing the platform name's index, not the user-friendly version, like 'squarespace' instead of 'Squarespace'. I think we should map them to the user-friendly name. Maybe we can reuse [this](https://github.com/Automattic/wp-calypso/blob/f36d18e84f1881e5646a118997091787b2c91da2/client/blocks/import/util.ts#L18-L33)?
3. 
<img width="856" alt="Screenshot 2024-10-15 at 7 01 58 AM" src="https://github.com/user-attachments/assets/e49a303a-b986-49b6-bd53-895d292bce56">

3. If the platform is Wix, it doesn't take me to an importer, rather it shows a screen with a loader indicating import has already started, without me providing any backup file or anything like that-
<img width="1404" alt="Screenshot 2024-10-15 at 7 06 12 AM" src="https://github.com/user-attachments/assets/22c0bbcf-f0ca-48e8-9379-e93fda57e93f">

gabrielcaires committed:
Merge branch 'trunk' into add/other-step

gabrielcaires committed:
Remove uncessary alias

gabrielcaires committed:
Use user friendly platform name

gabrielcaires committed:
Add support to undentieid platform

gabrielcaires committed:
simplify logic

gabrielcaires committed:
Move title to a const

gabrielcaires committed:
Fix back button

gabrielcaires commented:
> 1. The back button doesn't work on the this page -

Fixed [babdab4](https://github.com/Automattic/wp-calypso/pull/95246/commits/babdab43c833a71b66869331176eacdfff8b5fd7) 

> We are showing the platform name's index, not the user-friendly version, like 'squarespace' instead of 'Squarespace'.

Fixed [e333b6b](https://github.com/Automattic/wp-calypso/pull/95246/commits/e333b6b0566816559bce2d9584cf5c6424d94748)

I also added support to show a different message when we can't identify the platform

Imran92 commented:
> > If the platform is Wix, it doesn't take me to an importer, rather it shows a screen with a loader indicating import has already started, without me providing any backup file or anything like that-
> 
> I don't have the context, but it is intentional in all places we are running importers
> 
> https://github.com/Automattic/wp-calypso/blob/23fdaae6d35a5a1b756cd1bfc10219d36776edc6/client/landing/stepper/declarative-flow/internals/steps-repository/import/helper.ts#L34-L38

That's interesting 🤔 Out of curiosity, I checked our `importList` step, and it actually doesn't have `Wix` in the list, so I think we don't actually allow importing Wix site during the flow intentionally, otherwise it'd have been in that list. Also, showing an infinite loading screen to a paid user doesn't sound very good to me. So just as we show the "I need help" button to users who are on "Unknown" platform, should we do the same for Wix as well?

cc: @fditrapani 

<img width="893" alt="Screenshot 2024-10-16 at 10 40 06 AM" src="https://github.com/user-attachments/assets/ad36e34d-f6f5-4c95-89ee-c58f9101a155">

Also, as per our discussion here (p1729052919758479/1728407315.083759-slack-C07G7FG6J8Y), can we send a request to the backend API with `bypass_verification` set to `true` to create the ticket before we move the user to the `Mix-up` step?

gabrielcaires committed:
Create tickets when site is wordpress or using wpcom

gabrielcaires committed:
Remove unused hook properties

gabrielcaires committed:
Merge branch 'trunk' into add/other-step

gabrielcaires commented:
> Also, as per our discussion here (p1729052919758479/1728407315.083759-slack-C07G7FG6J8Y), can we send a request to the backend API with bypass_verification set to true to create the ticket before we move the user to the Mix-up step?

Updated [ab15ca6](https://github.com/Automattic/wp-calypso/pull/95246/commits/ab15ca63ee52e61d9cd8859fcf4a06201eafb32f)


>  in the list, so I think we don't actually allow importing Wix site during the flow intentionally; otherwise, it'd have been in that list. Also, showing an infinite loading screen to a paid user could sound better to me. 

Yes, definitely.  I suggest handling it in a separate ticket because all of our importer's flows use the same list and code we use here. It is more like a global list of supported importers.

gabrielcaires committed:
Merge branch 'trunk' into add/other-step

Imran92 reviewed (approved):
LGTM!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16935281

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @gabrielcaires for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16935281).
